File: app/src/main/java/com/example/android_challenge_btg/models/responses/MoviesResponse.java
package com.example.android_challenge_btg.models.responses;

import com.example.android_challenge_btg.models.Movie;

import java.math.BigInteger;
import java.util.List;

public class MoviesResponse {
    private int page;
    private BigInteger totalResults;
    private BigInteger totalPages;
    private List<Movie> results;

    public MoviesResponse(int page, BigInteger totalResults, BigInteger totalPages, List<Movie> results) {
        this.page = page;
        this.totalResults = totalResults;
        this.totalPages = totalPages;
        this.results = results;
    }

    public int getPage() {
        return page;
    }

    public void setPage(int page) {
        this.page = page;
    }

    public BigInteger getTotalResults() {
        return totalResults;
    }

    public void setTotalResults(BigInteger totalResults) {
        this.totalResults = totalResults;
    }

    public BigInteger getTotalPages() {
        return totalPages;
    }

    public void setTotalPages(BigInteger totalPages) {
        this.totalPages = totalPages;
    }

    public List<Movie> getResults() {
        return results;
    }

    public void setResults(List<Movie> results) {
        this.results = results;
    }
}

File: app/src/main/java/com/uchoa/btg/movie/models/api/MoviesResponse.kt
package com.uchoa.btg.movie.models.api

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName
import com.uchoa.btg.movie.models.Movie

class MoviesResponse {

    @SerializedName("page")
    @Expose
    var page: Int = 0

    @SerializedName("total_results")
    @Expose
    var totalResults: Int = 0

    @SerializedName("results")
    @Expose
    var movies: MutableList<Movie>? = null

    @SerializedName("total_pages")
    @Expose
    var totalPages: Int = 0
}